### PR TITLE
[CARBONDATA-3385]consider the merge index file also during clean files for compatibility case

### PR DESCRIPTION
**Pull request by akashrn5**
Merge index files are not considered during clean files in case of old stores.

Consider merge index files also for deleting during clean files.

Be sure to do all of the following checklist to help us incorporate 
your contribution quickly and easily:

 - [ ] Any interfaces changed?
 
 - [ ] Any backward compatibility impacted?
 
 - [ ] Document update required?

 - [ ] Testing done
        Please provide details on 
        - Whether new unit test cases have been added or why no new tests are required?
        - How it is tested? Please attach test report.
        - Is it a performance related change? Please attach the performance test report.
        - Any additional information to help reviewers in testing this change.
       
 - [ ] For large changes, please consider breaking it into sub-tasks under an umbrella JIRA.

### CODE REVIEW

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.1.0, Please check CI http://136.243.101.176:8080/job/ApacheCarbonPRBuilder2.1/3193/

File: core/src/main/java/org/apache/carbondata/core/util/DeleteLoadFolders.java [hunk 1]
```diff
@@ -128,7 +128,8 @@ private static void physicalFactAndMeasureMetadataDeletion(CarbonTable carbonTab
 
                 @Override public boolean accept(CarbonFile file) {
                   return (CarbonTablePath.isCarbonDataFile(file.getName()) ||
-                      CarbonTablePath.isCarbonIndexFile(file.getName()));
+                      CarbonTablePath.isCarbonIndexFile(file.getName()) ||
+                      CarbonTablePath.isCarbonMergeIndexFile(file.getName()));
```
Comment (KanakaKumar) [inline core/src/main/java/org/apache/carbondata/core/util/DeleteLoadFolders.java]: same may needs to be handled in org.apache.carbondata.core.metadata.SegmentFileStore#getSegmentFileForPhysicalDataPartitions

Please check

Comment (akashrn5) [reply core/src/main/java/org/apache/carbondata/core/util/DeleteLoadFolders.java]: actully for isIndexFile method, we are checking for startswith, so it considers merge index file also, as file extension is .carbonindexmerge. So this change is not required may be.

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.3.2, Please check CI http://136.243.101.176:8080/job/carbondataprbuilder2.3/11459/

**Comment by CarbonDataQA** [comment]
Build Success with Spark 2.2.1, Please check CI http://95.216.28.178:8080/job/ApacheCarbonPRBuilder1/3390/

File: core/src/main/java/org/apache/carbondata/core/util/DeleteLoadFolders.java (hunk 1 above)
Comment (KanakaKumar) [reply core/src/main/java/org/apache/carbondata/core/util/DeleteLoadFolders.java]: OK got it

**Pull request closed without merging**